blazar/manager/enforcement.py

```diff
@@ -196,6 +196,10 @@ def check_lease_duration(self, lease_values, lease=None):
                 prolong_allowed_from = lease['end_date'] - prolong_window
                 if (now >= prolong_allowed_from):
                     lease_duration = end_date - now
+                else:
+                    raise common_ex.NotAuthorized(
+                        'Prolong lease before the last %d seconds is not allowed' %
```

Comment:
Can we make this a bit more user-friendly? Suggest different text and a way to make the numbers easier to parse.

```suggestion
                        'Prolonging a lease is only allowed within its final %s' %
```

```python
def human_friendly_timedelta(seconds):
    delta = datetime.timedelta(seconds=seconds)
    # Prefer rough days/hours if we're on that time scale
    for attr in ['days', 'hours', 'minutes']:
        value = getattr(delta, attr)
        if value:
            return '%d %ss' % value, attr
    return '%d seconds' % seconds
```

Reply:
`timedelta` doesn't have attributes "hours" and "minutes". so I modified the function a little. Also, I'm not sure about this. For example, what if it's 2.5 days, the message will show as 2 days, which somehow incorrect. But I guess we will always go with int days/hours/minutes, instead of weird times like 101 seconds.